content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,235 @@
+and asynchronous (async) code in JavaScript.
+
+## Steps
+
+### 1. Hello XRP Ledger
+
+**Full code for this step:** [`1_hello.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/1_hello.js).
+
+Our first step is to write a "Hello World" like application, that interacts on a very basic level with the XRP Ledger 
+and displays some high-level information about the current ledger state on the screen. Nothing too fancy so far, as we 
+will take care of styling and GUI related coding in a later step:
+
+![Screenshot: Step 1, hello world equivalent](img/javascript-wallet-1.png)
+
+In the package.json file you can find prepared commands to run our application according to the steps comprising the 
+structure of this tutorial. To get the application running at this early stage of development, run the following command: 
+
+```console
+npm run hello
+```
+
+The source code of this step is as follows, and we will use this lightweight example to get privy on how the data flow 
+in electron works: 
+
+{{ include_code("_code-samples/build-a-wallet/js/1_hello.js", language="js") }}
+
+The main parts are a helper function that executes a ledger request and the main function that creates the application 
+window, calls our helper function and distributes its contents by broadcasting an event that will get picked up by the 
+frontend.
+
+We will use this example to study how to do Inter Process Communication (IPC) in Electron. For reasons of due diligence it 
+has to be mentioned that JavaScript has no real Processes / Threading because follows event-driven paradigm. Nonetheless 
+Electron provides us with two IPC modules called ipcMain and ipcRenderer. We can roughly equate those to the backend
+and the frontend when we think in terms of client-server applications. It works as follows:
+
+1. Create a function that enables the frontend to subscribe to backend events (in 1_preload.js)
+2. Make the function available by preloading it (webPreferences.preload during window creation)
+3. Use that function in the frontend (e.g. 1_renderer.js, loaded in 1_hello.html) to attach a callback that handles frontend updates when the event is dispatched
+4. Dispatch the event from the backend (e.g. appWindow.webContents.send('update-ledger-index', value))
+
+### 2.A. Show Ledger Updates by Polling
+
+**Full code for this step:** [`2_async-poll.js`]({{target.github_forkurl}}/tree/{{target.github_branch}}/content/_code-samples/build-a-wallet/js/2_async-poll.js).
+
+Our "Hello Ledger" application from Step 1. so far only shows the latest validated ledger sequence at the time when we 
+opened it. So let's take things up a notch and add some dashboard like functionality where our wallet app will keep in 
+sync with the ledger and display the latest specs and stats like a clock that is keeping track on time:
+
+![Screenshot: Step 2, show ledger updates](img/javascript-wallet-2.png)
+
```

Comment:
Adding in the new changes we're making related to the XRPL (Provided as an example of what you could do - I think context like this would be important to add to each section from here on out)

The screenshot of what the completed step looks like is very helpful for understanding what we're accomplishing :)
```suggestion

To accomplish this, we'll need to subscribe to the ledger using this snippet of code: 

<Code snippet>

When you subscribe to the `ledger` stream, anytime there is a new validated ledger your code will be updated...

```

Reply:
Ah I got a bit confused here - I think that context is still important, but I didn't realize that 2A/2B were replacements for each other. My recommendation would be to start with subscriptions and include the polling option as a small "Alternatively, you can also set this up using polling instead of subscriptions like this: <Code>" in order to keep a straightforward flow during the tutorial. 

(I don't know how important polling is as a technique relative to subscription - I think just showing one gets the message across without confusing people but open to hearing if you think otherwise)